src/components/avatar/Avatar.tsx

```diff
@@ -151,7 +152,8 @@ const Avatar = ({
   );
 
   const enableTooltip = tooltip && typeof fullName === 'string' && fullName.length > 0;
-  const Component = enableTooltip ? TooltippedBox : Box;
+  // @ts-ignore TS complains about non-overlapping props
+  const Component: GenericComponent<BoxProps | TooltipProps> = enableTooltip ? TooltippedBox : Box;
```

Comment:
PP: It seems the root of the problem is the type of `tooltipProps`. I prefer casting over using `@ts-ignore`. It seems like for all the components in this commit you can solve it by removing the `@ts-ignore` and `GenericComponent<BoxProps | TooltipProps> `
and when spreading the tooltipProps `{...(tooltipProps as TooltipProps)}`